Update account settings

trowland10 opened the pull request:
changed sql queries and added an e-mail check, and removed preferences from the dropdown menu. Deleted preferences.handlebars file

trowland10 committed:
Preferences Page

Updated the preferences page, added links to all the relevant pages, added log out button at bottom of page, And can update database in the account settings page.
To do: figure out what to do when you reload the page after updating the account settings.

trowland10 committed:
changes to post method

Made some changes to post method for update acocunt

trowland10 committed:
Changes to preferences page

added help to navbar and changed the preferences page around.

trowland10 committed:
Changed My Profile

Added link to update account settings at the bottom of the users profile

trowland10 committed:
Merge branch 'master' into tom

trowland10 committed:
Updates to account settings

Addresses some of the comments from previous pull request

mkoeplin commented on routes/index.js:
```diff
@@ -324,19 +322,26 @@ router.post('/updateaccount', common.isAuthenticated, (req, res, next) => {
-            res.render('auth/updateaccount', data)
+        if (results.length != 0 && results[0].id != req.session.userId) {
+            req.flash("error", "This email is already associated with another account")
+            res.reload();  //this might need to be changed
```
If the email already exists you won't want to do a reload as you'll clear out all the data they already entered. You should be using res.render here and passing back the data object you have. Typically in this case I would se the data.email to blank before sending back so it clears that field but leaves the rest. I'd also put back the title in the data so you're returning and still have the tab id or page title which should be using that as well.

trowland10 replied:
my new commit reflects these changes.

mkoeplin reviewed (changes requested):
Let's fix the return if the email already exists to send back the data they entered so they don't have to re-enter any other changes. See my notes on the file. We should also always be sending in a data.title when we return to a page.

trowland10 committed:
Changes to email error

Update account will now render with submitted data if an error with the email pops up

mkoeplin reviewed (approved):
Thanks for making that change!

halekyl reviewed (approved):
I don't see any issues, looks good to go.
